homeassistant/components/triggercmd/switch.py

```diff
@@ -0,0 +1,105 @@
+"""Platform for switch integration."""
+
+from __future__ import annotations
+
+import logging
+
+from triggercmd import client
+
+from homeassistant.components.switch import SwitchEntity
+from homeassistant.core import HomeAssistant
+from homeassistant.helpers.device_registry import DeviceInfo
+from homeassistant.helpers.entity_platform import AddEntitiesCallback
+
+from . import TriggercmdConfigEntry
+from .const import DOMAIN
+
+_LOGGER = logging.getLogger(__name__)
+
+
+async def async_setup_entry(
+    hass: HomeAssistant,
+    config_entry: TriggercmdConfigEntry,
+    async_add_entities: AddEntitiesCallback,
+) -> None:
+    """Add switch for passed config_entry in HA."""
+    hub = config_entry.runtime_data.hub
+    async_add_entities(TRIGGERcmdSwitch(switch) for switch in hub.switches)
+
+
+class TRIGGERcmdSwitch(SwitchEntity):
+    """Representation of a Switch."""
+
+    should_poll = False
+
+    def __init__(self, switch) -> None:
+        """Initialize the switch."""
+        self._switch = switch
+        self._name = switch.name
+        self._state = False
+        self._attr_unique_id = f"{self._switch.switch_id}_switch"
+
+        self._attr_name = self._switch.name
+
+    async def async_added_to_hass(self) -> None:
+        """Run when this Entity has been added to HA."""
+        self._switch.register_callback(self.async_write_ha_state)
+
+    async def async_will_remove_from_hass(self) -> None:
+        """Entity being removed from hass."""
+        self._switch.remove_callback(self.async_write_ha_state)
+
+    @property
+    def device_info(self) -> DeviceInfo:
```

Comment:
I'm not familiar with triggercmd but from the variable names I think we should create a device per computer

Reply:
I considered that, but I decided each triggercmd trigger should be represented as a device because the Alexa, Google Assistant, and SmartThings integrations create a virtual on/off switch device for each trigger.  It makes it easier to educate triggercmd users if the integrations work similarly.  I hope that's ok.

Reply:
Nope we should create one device per computer so all triggers of one computer are grouped in the UI too